terraform/eks-cloudwatch/appmesh/appmesh.tf

```diff
@@ -0,0 +1,186 @@
+# ------------------------------------------------------------------------
+# Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+# -------------------------------------------------------------------------
+
+variable "provider_url" {
+  type    = string
+  default = ""
+}
+
+variable "sample_app_image_repo" {
+  type    = string
+  default = null
+}
+
+variable "testing_id" {
+  type    = string
+  default = ""
+}
+
+variable "region" {
+  type    = string
+  default = "us-west-2"
+}
+
+variable "testcase" {
+  type    = string
+  default = "../testcases/container_insight"
+}
+
+variable "kubeconfig" {
+  type = string
+  default = "kubeconfig"
+}
+
+output "metric_dimension_namespace" {
+  value = kubernetes_namespace.traffic_ns.metadata[0].name
+}
+
+data "aws_caller_identity" "current" {}
+
+locals {
+  mesh_name      = "howto-k8s-http-headers-${var.testing_id}"
+  aws_account_id = data.aws_caller_identity.current.account_id
+}
+
+data "aws_iam_policy" "appmesh_policy" {
+  arn = "arn:aws:iam::${local.aws_account_id}:policy/AWSAppMeshK8sControllerIAMPolicy"
+}
+
+module "iam_assumable_role_with_oidc" {
+  source       = "terraform-aws-modules/iam/aws//modules/iam-assumable-role-with-oidc"
+  version      = "v3.13.0"
```

Comment:
should we use version constraint to get the latest version? eg `"~> 3.0"`

Reply:
Done